### PR TITLE
Fix data cite issues and code snippets

### LINKED ISSUE #14: Update href to data-cite
All links to other specs made via `href` should be replaced with `data-cite` as explained in https://github.com/w3c/rdf-star-wg/wiki/Editor's-guide#citing-other-documents-and-definitions.

### PR DESCRIPTION
**Pull request by rubensworks**
Closes #14

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/sparql-query/pull/45.html" title="Last updated on Apr 7, 2023, 10:43 AM UTC (0e8b8bd)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/sparql-query/45/e3b9d77...0e8b8bd.html" title="Last updated on Apr 7, 2023, 10:43 AM UTC (0e8b8bd)">Diff</a>

### CODE REVIEW

File: spec/index.html [hunk 1]
```diff
@@ -895,28 +895,28 @@ <h3>Blank Node Labels in Query Results</h3>
       </section>
       <section id="CreatingValuesWithExpressions">
         <h3>Creating Values with Expressions</h3>
-        <p>SPARQL 1.1 allows to create values from complex expressions. The queries below show how to
+        <p>SPARQL 1.2 allows to create values from complex expressions. The queries below show how to
```
Comment (afs) [inline spec/index.html:898]:
```suggestion
        <p>SPARQL 1.2 allows values to be created from complex expressions. The queries below show how to
```

File: spec/index.html [hunk 2]
```diff
@@ -977,8 +977,8 @@ <h3>Building RDF Graphs</h3>
 </pre>
             </div>
           </div>
-          <p>which can be serialized in <a href=
-                                           "http://www.w3.org/TR/rdf-syntax-grammar/">RDF/XML</a> as:</p>
+          <p>which can be serialized in <a data-cite=
+                                           "RDF12-XML#">RDF/XML</a> as:</p>
```
Comment (afs) [inline spec/index.html:981]:
```suggestion
          <p>which can be serialized in <a data-cite="RDF12-XML#">RDF/XML</a> as:</p>
```

**Review by afs — CHANGES_REQUESTED** [review]
2 small suggestions but otherwise LGTM.

**Review by hartig — APPROVED** [review]

File: spec/index.html [hunk 3]
```diff
@@ -895,7 +895,7 @@ <h3>Blank Node Labels in Query Results</h3>
       </section>
       <section id="CreatingValuesWithExpressions">
         <h3>Creating Values with Expressions</h3>
-        <p>SPARQL 1.2 allows to create values from complex expressions. The queries below show how to
+        <p>SPARQL 1.2 allows values to be created from complex expressions. The queries below show how to
```
Comment (TallTed) [inline spec/index.html:898]:
```suggestion
        <p>SPARQL 1.2 allows values to be created from complex expressions. The queries below show how
```

**Review by afs — APPROVED** [review]

**Review by domel — APPROVED** [review]

**Pull request merged**